packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -0,0 +1,351 @@
+        }
+        assert.equal(testObject.root.get("foo"), "bar");
+    }
+}
+
+describeNoCompat("Batching failures", (getTestObjectProvider) => {
+    it("working proxy",
+    async function() {
+        const provider = getTestObjectProvider(true);
+
+        const proxyDsf = createFunctionOverrideProxy<IDocumentServiceFactory>(
+            provider.documentServiceFactory,
+            {
+                createDocumentService: {
+                    connectToDeltaStream: {
+                        submit: (ds) => (messages) => {
+                            // validate a no-op proxy works
+                            ds.submit(messages);
+                        },
+                    },
+                },
+            });
+        await runAndValidateBatch(provider, proxyDsf, this.timeout());
+    });
+    it("working batch",
+    async function() {
+        const provider = getTestObjectProvider(true);
+        await runAndValidateBatch(provider, provider.documentServiceFactory, this.timeout());
+    });
+    describe("client sends invalid batches ", () => {
+        itExpects("Batch end without start",
+        [
+            { eventName: "fluid:telemetry:Container:ContainerClose", error: "OpBatchIncomplete" },
+        ],
+        async function() {
+            const provider = getTestObjectProvider(true);
+
+            const proxyDsf = createFunctionOverrideProxy<IDocumentServiceFactory>(
+                provider.documentServiceFactory,
+                {
+                    createDocumentService: {
+                        connectToDeltaStream: {
+                            submit: (ds) => (messages) => {
+                                const newMessages = [...messages];
+                                const batchStartIndex = newMessages.findIndex((m) => m.metadata?.batch === true);
+                                if (batchStartIndex >= 0) {
+                                    newMessages[batchStartIndex] = {
+                                        ... newMessages[batchStartIndex],
+                                        metadata: {
+                                            ... newMessages[batchStartIndex].metadata,
+                                            batch: undefined,
+                                        },
+                                    };
+                                }
+                                ds.submit(newMessages);
+                            },
+                        },
+                    },
+                });
+
+            try {
+                await runAndValidateBatch(provider, proxyDsf, this.timeout());
+                assert.fail("expected error");
+            } catch (e) {
+                assert(isILoggingError(e), `${e}`);
+                assert.equal(e.message, "OpBatchIncomplete", e);
+            }
+        });
+
+        // bug bug: container runtime never unpauses if there is no batch end
+        itExpects.skip("Batch start without end",
```

Comment:
nit: use `it` instead of `itExpects` where an error is not expected.

Reply:
i left it ItExpect because if we fix it there should be an error in this case rather than hang forever